plugins/parser-ingestion/src/main/scala/com/stratio/sparkta/plugin/parser/ingestion/IngestionParser.scala

```diff
@@ -0,0 +1,110 @@
+                      properties: Map[String, JSerializable])
+  extends Parser(name, order, inputField, outputFields, properties) {
+
+  private val DatetimePattern = "yyyy-MM-dd HH:mm:ss"
+
+  override def parse(data: Event): Event = {
+    val ingestionModel = JSON.parseFull(data.keyMap.get(inputField).get.asInstanceOf[String])
+      .get.asInstanceOf[Map[String,JSerializable]]
+    val columnList = ingestionModel.get("columns").get.asInstanceOf[List[Map[String, String]]]
+    val columnPairs = extractColumnPairs(columnList)
+    val allParsedPairs = parseWithSchema(columnPairs, Map())._2
+    val filteredParsedPairs = allParsedPairs.filter(element => outputFields.contains(element._1))
+    new Event(filteredParsedPairs)
+  }
+
+  // XXX Private methods.
+
+  private def extractColumnPairs(columnList: List[Map[String, String]]): List[(String, String)] = {
+    val columnListKeyValue = for {
+      columnElement <- columnList
+      value <- columnElement
+    } yield Map(value._1 -> value._2)
+    extractColumnPairElement(columnListKeyValue, List())._2
+  }
+
+  @tailrec
+  private def extractColumnPairElement(columnList: List[Map[String, String]],
+                                       result: List[(String,String)])
+  : (List[Map[String,String]], List[(String,String)]) = {
+    if(columnList.isEmpty) {
+      (columnList, result)
+    } else {
+      val currentValue = columnList.last.head._2
+      val columnListWithoutValue = columnList.init
+      val currentKey = columnListWithoutValue.last.head._2
+      val columnListWithoutKeyAndValue = columnListWithoutValue.init
+      extractColumnPairElement(columnListWithoutKeyAndValue, result.:::(List((currentKey, currentValue))))
+    }
+  }
+
+  @tailrec
+  private def parseWithSchema(elementList: List[(String, String)],
+                              currentMap: Map[String,JSerializable])
+  : (List[(String, JSerializable)],Map[String,JSerializable]) = {
+    if(elementList.isEmpty) {
+      (elementList, currentMap)
+    } else {
+      val currentElement = elementList.last
+      val newElementList = elementList.init
+      val newCurrentMap = currentMap ++ parseElementWithSchema(currentElement)
+      parseWithSchema(newElementList, newCurrentMap)
+    }
+  }
+
+  private def parseElementWithSchema(element: (String, JSerializable)): Map[String, JSerializable] = {
+    val key = element._1
+    val value = element._2.toString
+    val dataType = (properties.get(element._1) match {
+      case Some(value) => value.toString
+      case _ => throw new IllegalArgumentException(s"The element $element do not exists in the schema.")
+    }).toLowerCase
+
+    dataType match {
+      case "long" =>
+        Map(key -> value.toLong)
+      case "int" | "integer" =>
+        Map(key -> value.toInt)
+      case "string" =>
+        Map(key -> value)
+      case "float" =>
+        Map(key -> value.toFloat)
+      case "double" =>
+        Map(key -> value.toDouble)
+      case "datetime" =>
```

Comment:
Please remove this case. This will type have to be parsed by DateTime parser.

It could be confusing for the user since format '"yyyy-MM-dd HH:mm:ss"' is a Hive format used in the example as a string. It is not an internal type of Stratio Ingestion